### PR TITLE
Force references to libcrypto functions when linking statically

### PR DESCRIPTION
**Pull request by justinboswell**
prevents dead-stripping

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/unix/openssl_platform_init.c [hunk 1]
```diff
@@ -15,6 +15,39 @@ static struct openssl_evp_md_ctx_table evp_md_ctx_table;
...
+
+/* common */
+extern int HMAC_Update(HMAC_CTX *, const unsigned char *, size_t) __attribute__((weak)) __attribute__((used));
+extern int HMAC_Final(HMAC_CTX *, unsigned char *, unsigned int *) __attribute__((weak)) __attribute__((used));
+extern int HMAC_Init_ex(HMAC_CTX *, const void *, int, const EVP_MD *, ENGINE *) __attribute__((weak))
+__attribute__((used));
+
+/* EVP_MD_CTX API */
+/* 1.0.2 */
+/*extern EVP_MD_CTX *EVP_MD_CTX_create(void) __attribute__((weak)) __attribute__((used));*/
```
Comment (bretambrose) [inline source/unix/openssl_platform_init.c:38]: Delete?

File: source/unix/openssl_platform_init.c [hunk 2]
```diff
@@ -47,14 +80,15 @@ static void s_hmac_ctx_free(HMAC_CTX *ctx) {
...
-static void s_hmac_ctx_reset(HMAC_CTX *ctx) {
+static int s_hmac_ctx_reset(HMAC_CTX *ctx) {
     AWS_PRECONDITION(ctx);
     AWS_PRECONDITION(
         g_aws_openssl_hmac_ctx_table->init_fn != s_hmac_ctx_init_noop &&
         g_aws_openssl_hmac_ctx_table->clean_up_fn != s_hmac_ctx_clean_up_noop &&
         "libcrypto 1.0 reset called on libcrypto 1.1 vtable");
     g_aws_openssl_hmac_ctx_table->clean_up_fn(ctx);
     g_aws_openssl_hmac_ctx_table->init_fn(ctx);
+    return true;
```
Comment (bretambrose) [inline source/unix/openssl_platform_init.c:91]: I'd change this to a 1 with a note that it matches the semantics of HMAC_CTX_reset()

**Review by bretambrose — APPROVED** [review]

File: source/unix/openssl_platform_init.c (hunk 2 above)
Comment (justinboswell) [reply source/unix/openssl_platform_init.c:91]: Can do

File: source/unix/openssl_platform_init.c (hunk 1 above)
Comment (justinboswell) [reply source/unix/openssl_platform_init.c:38]: there may have been some flailing, I thought I caught it all

**Pull request merged**